crates/solver/src/solver/http_solver.rs

```diff
@@ -284,8 +287,16 @@ fn order_models(
                 Exchange::ZeroEx => gas_model.zeroex_order_cost(),
             };
 
+            let uid = match OrderUid::from_str(&order.id) {
+                Ok(uid) => uid,
```

Comment:
I think we might have to think more carefully about `0x` liquidity orders here. Right now the `id` field gets [populated](https://github.com/cowprotocol/services/blob/main/crates/solver/src/liquidity/zeroex.rs#L114) with an `order_hash` so `OrderUid::from_str()` should not fail but it might result in malformed or nonsensical `OrderUid`s.
This is probably good enough if we want to identify orders uniquely across different auctions but if people expect to find every `OrderUid` of an auction in our order book they would be mistaken.

Reply:
Thanks for mentioning this as a special case (I thought `OrderUid::from_str` is safe always). Anyway, I don't think we can do anything to mitigate the `0x` difference in id creation. I was considering to hash by `String` instead of `OrderUid` but `OrderUid` seemed more expressive.

Reply:
> I thought OrderUid::from_str is safe always.

I think it can only fail if the `str` contains non-hex characters but that wasn't the problem anyway. Even if it succeeds the created `OrderUid` does not exist in our database.

> Anyway, I don't think we can do anything to mitigate the 0x difference in id creation. I was considering to hash orders ids as String instead of OrderUid but OrderUid seemed more expressive.

I agree that an `OrderUid` is to be preferred over a simple `String`. Maybe a comment explaining this issue somewhere in the code would already be good for now. If people are actually asking us why they can't find those ids in our order book we might have to consider marking those orders differently somehow.

Reply:
Makes sense. I've added the comment.